src/OSPSuite.Presentation/Presenters/SimulationOutputMappingPresenter.cs

```diff
@@ -75,18 +68,20 @@ public void Refresh()
       {
          this.DoWithinLatch(() =>
          {
-            UpdateCache();
+            updateOutputLists();
             updateOutputMappingList();
             _view.BindTo(_listOfOutputMappingDTOs);
          });
       }
 
-      public void UpdateCache()
+      private void updateOutputLists()
       {
          _allAvailableOutputs.Clear();
+         var outputs = _entitiesInSimulationRetriever.OutputsFrom(_simulation);
```

Comment:
this code was called in a loop for each output. The cache was not used at all really. I am sure we had discussed creating a cache to avoid calling in a loop so probably this was started but not finihsed